PWGCF/FemtoDream/Core/femtoDreamCollisionSelection.h

```diff
@@ -246,6 +313,236 @@ class FemtoDreamCollisionSelection
+
+    for (size_t i = 0; i < nCent; ++i) {
+      for (size_t j = 0; j < nBins; ++j) {
+        res[i][j] = flat[i * nBins + j];
+      }
+    }
+    return res;
+  }
+
+  /// \todo to be implemented!
+  /// Get the bin number of qn-vector(FT0C) of an event
+  /// \param centBinWidth centrality bin width, example: per 1%, per 10% ...
+  /// \return bin number of qn-vector of the event
+  int myqnBin(float centrality, float centMax, std::vector<float> qnBinSeparator, bool doFillHisto, float fSpher, float qn, const int numQnBins, float mult, float centBinWidth = 1.f)
+  {
+    auto twoDSeparator = getQnBinSeparator2D(qnBinSeparator, numQnBins);
+    if (twoDSeparator.empty() || twoDSeparator[0][0] == -999.) {
+      LOGP(warning, "ConfQnBinSeparator not set, using default fallback!");
+      return -999; // safe fallback
+    }
+
+    if (doFillHisto)
+      mHistogramQn->fill(HIST("Event/centFT0CBefore"), centrality);
+
+    int qnBin = -999;
+    int mycentBin = static_cast<int>(centrality / centBinWidth);
+    if (mycentBin >= static_cast<int>(centMax / centBinWidth))
+      return qnBin;
+
+    if (mycentBin > static_cast<int>(twoDSeparator.size()) - 1)
+      return qnBin;
+
+    for (int iqn(0); iqn < static_cast<int>(twoDSeparator[mycentBin].size()) - 1; ++iqn) {
+      if (qn > twoDSeparator[mycentBin][iqn] && qn <= twoDSeparator[mycentBin][iqn + 1]) {
+        qnBin = iqn;
+        break;
+      } else {
+        continue;
+      }
+    }
+
+    mQnBin = qnBin;
+
+    if (doFillHisto) {
+      mHistogramQn->fill(HIST("Event/centFT0CAfter"), centrality);
+      mHistogramQn->fill(HIST("Event/centVsqn"), centrality, qn);
+      mHistogramQn->fill(HIST("Event/centVsqnVsSpher"), centrality, qn, fSpher);
+      mHistogramQn->fill(HIST("Event/qnBin"), qnBin);
+      if (qnBin >= 0 && qnBin < numQnBins) {
+        switch (qnBin) {
+          case 0:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("0"), mult);
+            break;
+          case 1:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("1"), mult);
+            break;
+          case 2:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("2"), mult);
+            break;
+          case 3:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("3"), mult);
+            break;
+          case 4:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("4"), mult);
+            break;
+          case 5:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("5"), mult);
+            break;
+          case 6:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("6"), mult);
+            break;
+          case 7:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("7"), mult);
+            break;
+          case 8:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("8"), mult);
+            break;
+          case 9:
+            mHistogramQn->fill(HIST("Qn/mult_") + HIST("9"), mult);
+            break;
```

Comment:
If you store the histogram pointer in a vector the switch will not be needed as you will access to the proper histogram with the bin as index to the vector